src/content/services/monetizationLinkManager.ts

```diff
@@ -30,14 +32,19 @@ export class MonetizationLinkManager extends EventEmitter {
     { walletAddress: WalletAddress; requestId: string }
   >();
 
-  constructor({ window, document, logger, message }: Cradle) {
-    super();
+  constructor({ document, logger, message, global }: Cradle) {
     Object.assign(this, {
-      window,
+      global,
       document,
       logger,
       message,
+      window: global.window,
     });
+    const { MutationObserver, crypto, window } = this.global;
+
+    this.id = crypto.randomUUID();
+    this.isTopFrame = window === window.top;
+    this.isFirstLevelFrame = window.parent === window.top;
```

Comment:
Moved these up otherwise mutation observer could be fired before these are assigned?! Saw happening in tests only.